File: internal/model/request/register_request.go
package request

// RegisterRequest 注册请求结构体
type RegisterRequest struct {
	Username    string `json:"username" binding:"required,min=2,max=20"`                      // 用户名
	Password    string `json:"password" binding:"required,min=6"`                             // 密码
	PhoneNumber string `json:"phoneNumber,omitempty" binding:"omitempty,startswith=1,len=11"` // 电话号码
	Email       string `json:"email,omitempty" binding:"omitempty,email"`                     // 邮箱地址
}

// RegisterRequestValidationMessages 注册请求验证消息
var RegisterRequestValidationMessages = map[string]string{
	"Username.required":      "用户名不能为空",
	"Username.min":           "用户名长度应在2-20个字符之间",
	"Username.max":           "用户名长度应在2-20个字符之间",
	"Password.required":      "密码不能为空",
	"Password.min":           "密码至少6位",
	"PhoneNumber.startswith": "电话号码格式不正确",
	"PhoneNumber.len":        "电话号码格式不正确",
	"Email.email":            "邮箱格式不正确",
}

File: backend/src/main/java/com/bryan/system/domain/request/auth/RegisterRequest.java
package com.bryan.system.domain.request.auth;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Getter;

/**
 * 用户注册请求对象
 *
 * @author Bryan Long
 */
@Getter
public class RegisterRequest {
    /**
     * 用户名。
     * 在注册时是必需的。
     */
    @NotBlank(message = "用户名不能为空")
    @Size(min = 2, max = 20, message = "用户名长度应在2-20个字符之间")
    private String username;

    /**
     * 密码。
     * 在注册时是必需的。
     */
    @NotBlank(message = "密码不能为空")
    @Size(min = 6, message = "密码至少6位")
    private String password;

    /**
     * 电话号码。
     * 在注册时是可选的，如果提供则添加，如果不提供则不添加。
     */
    @Pattern(regexp = "^1[3-9]\\d{9}$", message = "电话号码格式不正确")
    private String phone;

    /**
     * 邮箱地址。
     * 在注册时是可选的，如果提供则添加，如果不提供则不添加。
     */
    @Email(message = "邮箱格式不正确")
    private String email;
}
